### PR TITLE
fix: pkce does not generate truly random values in a browser

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?
As @dshukertjr pointed out, the current version of `generatePKCEVerifier()` has the random generation deleted. There should be an additional branch which calls `getRandomValues()` in order to truly set random values.

## What is the current behavior?
When the user is in a browser, the array defaults to the randomly initialized values as set by RAM which is not ideal.

## What is the new behavior?
Generate random values.

### CODE REVIEW

**Review by kangmingtay — APPROVED** [review]

**Review by dshukertjr — COMMENTED** [review]
Not directly related to this PR, but we can remove the `async` keyword on `generatePKCEVerifier` here, right?

**Comment by dshukertjr** [comment]
I think I saw few other places that we can remove async from. Let me create a PR

File: src/lib/helpers.ts [hunk 1]
```diff
@@ -249,6 +249,8 @@ export async function generatePKCEVerifier() {
     for (let i = 0; i < verifierLength; i++) {
       array[i] = Math.floor(Math.random() * 256)
     }
+  } else {
+    window.crypto.getRandomValues(array)
```
Comment (dshukertjr) [inline src/lib/helpers.ts:253]: Do we need to make `array` `var` and assign it here?
```suggestion
    array = window.crypto.getRandomValues(array)
```

Comment (dshukertjr) [reply src/lib/helpers.ts:253]: Ah, sorry, never mind. Just verified that the current code works!

**Comment by J0** [comment]
I think some of the `window` methods are async, so you might not be able to remove some of them but if you find any feel free to file a PR.  I'll also take a look later

**Pull request merged**